graphql/parser.go

```diff
@@ -368,18 +368,30 @@ func Parse(source string, vars map[string]interface{}) (*Query, error) {
 		return nil, NewClientError("must have a single query")
 	}
 
+	kind := queryDefinition.Operation
+	var name string
+	if queryDefinition.Name != nil {
+		name = queryDefinition.Name.Value
+	}
+
+	rv := &Query{
+		Name:         name,
+		Kind:         kind,
+		SelectionSet: nil,
+	}
+
 	// Parse variable definitions, default values, etc.
 	var defaultedVars map[string]interface{}
 	for _, variableDefinition := range queryDefinition.VariableDefinitions {
 		name := variableDefinition.Variable.Name.Value
 
 		if _, ok := variableDefinition.Type.(*ast.NonNull); ok {
 			if variableDefinition.DefaultValue != nil {
-				return nil, NewClientError("required varaible cannot provide a default value: $%s", name)
+				return rv, NewClientError("required variable cannot provide a default value: $%s", name)
```

Comment:
🎉

Reply:
We are sure that the function that calls Parse looks for non-nil error instead of nil result?

Reply:
yeah, we do this in other places downstream as wel